test-src/manageamiset_test.js

```diff
@@ -0,0 +1,144 @@
+var slugid = require('slugid');
+var assume = require('assume');
+var debug = require('debug')('test');
+var _ = require('lodash');
+var mock = require('./mock-amisets');
+var base = require('taskcluster-base');
+
+var makeAmiSet = mock.makeAmiSet;
+
+var main = require('../lib/main');
+var helper = require('./helper');
+
+describe('AMI Set api', () => {
+
+  var amiSetDefinition = makeAmiSet();
+  var amiSetChanged = _.clone(amiSetDefinition);
+  amiSetChanged.amis.push({
+    region: 'us-east-1',
+    hvm: 'ami-0123',
+    pv: 'ami-0345',
+  });
+
+  var id = amiSetDefinition.id;
+  var amis = amiSetDefinition.amis;
+
+  // testamiSetid will be used to create a second amiSet and test listAmiSets
+  var testamiSetid = 'amiset-2222';
+  var lastModified;
+
+  let amiSet;
+
+  let client;
+
+  before(async () => {
+    amiSet = await main('AmiSet', {process: 'AmiSet', profile: 'test'});
+
+    client = helper.getClient();
+  });
+
+  beforeEach(async () => {
+    await main('tableCleaner', {process: 'tableCleaner', profile: 'test'});
```

Comment:
We should add the `AmiSet` entity to the `tableCleaner` component in `src/main.js` so that this table is actually cleaned out.